Stripe/STPAPIClient.m

```diff
@@ -115,33 +116,35 @@ + (instancetype)sharedClient {
 }
 
 - (instancetype)init {
-    return [self initWithConfiguration:[STPPaymentConfiguration sharedConfiguration]];
-}
-
-- (instancetype)initWithPublishableKey:(NSString *)publishableKey {
-    STPPaymentConfiguration *config = [[STPPaymentConfiguration alloc] init];
-    config.publishableKey = [publishableKey copy];
-    return [self initWithConfiguration:config];
-}
-
-- (instancetype)initWithConfiguration:(STPPaymentConfiguration *)configuration {
-    NSString *publishableKey = [configuration.publishableKey copy];
-    if (publishableKey) {
-        [self.class validateKey:publishableKey];
-    }
     self = [super init];
     if (self) {
-        _apiKey = publishableKey;
         _apiURL = [NSURL URLWithString:APIBaseURL];
-        _configuration = configuration;
-        _stripeAccount = configuration.stripeAccount;
+        _configuration = [STPPaymentConfiguration sharedConfiguration];
         _sourcePollers = [NSMutableDictionary dictionary];
         _sourcePollersQueue = dispatch_queue_create("com.stripe.sourcepollers", DISPATCH_QUEUE_SERIAL);
         _urlSession = [NSURLSession sessionWithConfiguration:[self.class sharedUrlSessionConfiguration]];
+        _publishableKey = [Stripe defaultPublishableKey];
```

Comment:
copy

Reply:
actually wouldn't we not want to do this so we return the static _defaultPublishableKey?

Reply:
Good call, added a test to catch this.